MaxText/configs/base.yml

```diff
@@ -177,6 +177,7 @@ profiler_steps: 5
 # data_shuffle_seed and init_weights_seed (i.e. reproducible losses)
 enable_dropout: True
 enable_data_shuffling: True
+enable_file_shuffling: False
```

Comment:
Sorry I'm a little confused here? Did you add a new field for the same meaning as the above field?

Reply:
I added a new flag. These are shuffling at different levels. "enable_file_shuffling" correspondings to "shuffle_files" here: https://github.com/google/maxtext/blob/main/MaxText/input_pipeline.py#L183. "enable_data_shuffling" is used here: https://github.com/google/maxtext/blob/main/MaxText/input_pipeline.py#L133-L134. I separate these two flags because when enable_file_shuffling=True, "shards are shuffled for each epoch, so reading is not deterministic anymore" (https://www.tensorflow.org/datasets/determinism) and there's no way to set a fixed seed for file shuffle. While with "enable_data_shuffling=true", determinism can be achieved by setting data_shuffle_seed

Reply:
I added some comments to config, hopefully it's more clear now

Reply:
Thank you for the deep explanation. Are we sure we want to expose both of them?

Separately, if you are sure about this, the names/comments need to be crisper.

Reply:
I just took another look and found that shuffle_file can also be deterministic when read_config is set. Initially I saw my unit test for determinism was failing so I changed the flag. But I then realized the end-to-end determinism is passing and my unit test failure was due to not setting read_config. I revert the config change and fix unit tests now.